src/app/app_execution.rs

```diff
@@ -108,3 +136,188 @@ impl AppExecution {
         Ok(())
     }
 }
+
+/// A stream of [`RecordBatch`]es that can be paginated for display in the TUI.
+pub struct PaginatingRecordBatchStream {
+    // currently executing stream
+    inner: SendableRecordBatchStream,
```

Comment:
Because `SendableRecordBatchStream` is not `Sync` this struct doesnt work well with our `AppEvent` which requires `Send` and `Sync`.  I think we can restructure so that the batches / current idx are managed separate from the batch stream so they can be sent as `AppEvent` and state can be updated in the existing pattern (i.e in our `handlers`) and then we dont need to lock during rendering.